pkg/apis/broker/v2alpha4/activemqartemis_types.go

```diff
@@ -0,0 +1,200 @@
+package v2alpha4
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/olm"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ActiveMQArtemisSpec defines the desired state of ActiveMQArtemis
+// +k8s:openapi-gen=true
+type ActiveMQArtemisSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	AdminUser      string                  `json:"adminUser,omitempty"`
+	AdminPassword  string                  `json:"adminPassword,omitempty"`
+	DeploymentPlan DeploymentPlanType      `json:"deploymentPlan,omitempty"`
+	Acceptors      []AcceptorType          `json:"acceptors,omitempty"`
+	Connectors     []ConnectorType         `json:"connectors,omitempty"`
+	Console        ConsoleType             `json:"console,omitempty"`
+	Version        string                  `json:"version,omitempty"`
+	Upgrades       ActiveMQArtemisUpgrades `json:"upgrades,omitempty"`
+	//below are v2alpha4 types
+	AddressSettings AddressSettingsType `json:"addressSettings,omitempty"`
+}
+
+type AddressSettingsType struct {
+	ApplyRule      *string              `json:"applyRule,omitempty"`
+	AddressSetting []AddressSettingType `json:"addressSetting,omitempty"`
+}
+
+type AddressSettingType struct {
+	DeadLetterAddress                  *string `json:"deadLetterAddress,omitempty"`
+	AutoCreateDeadLetterResources      *bool   `json:"autoCreateDeadLetterResources,omitempty"`
+	DeadLetterQueuePrefix              *string `json:"deadLetterQueuePrefix,omitempty"`
+	DeadLetterQueueSuffix              *string `json:"deadLetterQueueSuffix,omitempty"`
+	ExpiryAddress                      *string `json:"expiryAddress,omitempty"`
+	AutoCreateExpiryResources          *bool   `json:"autoCreateExpiryResources,omitempty"`
+	ExpiryQueuePrefix                  *string `json:"expiryQueuePrefix,omitempty"`
+	ExpiryQueueSuffix                  *string `json:"expiryQueueSuffix,omitempty"`
+	ExpiryDelay                        *int32  `json:"expiryDelay,omitempty"`
+	MinExpiryDelay                     *int32  `json:"minExpiryDelay,omitempty"`
+	MaxExpiryDelay                     *int32  `json:"maxExpiryDelay,omitempty"`
+	RedeliveryDelay                    *int32  `json:"redeliveryDelay,omitempty"`
+	RedeliveryDelayMultiplier          *int32  `json:"redeliveryDelayMultiplier,omitempty"`
+	RedeliveryCollisionAvoidanceFactor *int32  `json:"redeliveryCollisionAvoidanceFactor,omitempty"`
```

Comment:
@gaohoward Needs to be float32 as per #31

Reply:
yes will update that